### ISSUE TITLE
Nexus 7 (flo) Error: systemimage: Error: Error: Failed to push file 0: Error: Push failed: out of space

### ISSUE THREAD

**Issue by Catbirby** [ubports/ubports-installer#1114]
*Automatically generated error report* 
UBports Installer Version: 0.4.14-beta 
Device: flo
OS to install: Ubuntu Touch
Settings: {channel:ubports-touch/16.04/edge,wipe:true,bootstrap:true,format:true}
Package: exe
Operating System: Microsoft Windows [Version 10.0.18363.592] ia32 
NodeJS version: v8.2.1 

Error log: https://paste.ubuntu.com//p/yDt5XPVt38/

To add some additional information and refer to older threads/issues I found on this. 
- I did revert back to the base OS and firmware before installing since I read that it could be an issues with updating to android P.
- I've tried this several other times in the past few years and always have had the same roadblock. 
- This time I thought the issue might be fixed from me repartitioning the device but it didn't change anything at all.
- This issue occurs when trying to install any version of UbTouch (I.E. Stable, rc, edge, devel)

Similar issue where this is said to be fixed is #615

**Comment by MrAdam** [comment]
I got the exact same issue

**Comment by chhager** [comment]
repartitioning back to stock configuration worked for me on my nexus 5.

**Comment by NeoTheThird** [comment]
Seems like there's more hardware variations in flo than we anticipated, this needs further investigation. No idea what's going on. But i do know that we fixed *some* of the flo issues. Annoying. Sorry about this, y'all.

**Comment by ChiefPete** [comment]
Here is memory/partition layout from my 16GB tablet:

flo/razor partitions Android 6.0.1 razor-mob30x ROM

root@flo:/ # blkid                                                         
/dev/block/loop0: UUID="57f8f4bc-abf4-655f-bf67-946fc0f9f25b" TYPE="ext4" 
/dev/block/mmcblk0p1: SEC_TYPE="msdos" LABEL="G0012300720" UUID="00BC-614E" TYPE="vfat" 
/dev/block/mmcblk0p4: UUID="57f8f4bc-abf4-655f-bf67-946fc0f9f25b" TYPE="ext4" 
/dev/block/mmcblk0p22: LABEL="system" UUID="da594c53-9beb-f85c-85c5-cedf76546f7a" TYPE="ext4" 
/dev/block/mmcblk0p23: UUID="4d747700-3cbe-43b9-b71c-0c00dd4b32ad" TYPE="ext4" 
/dev/block/mmcblk0p30: UUID="57f8f4bc-abf4-655f-bf67-946fc0f9f25b" TYPE="ext4" 
/dev/block/mmcblk0p31: UUID="18f901b3-c7a7-464c-9890-c7a0debab9c3" TYPE="ext4"


****************************************************************************************************


shell@flo:/ $ cat /proc/partitions
major minor  #blocks  name

 179        0   15388672 mmcblk0
 179        1      87536 mmcblk0p1
 179        2       3072 mmcblk0p2
 179        3       3072 mmcblk0p3
 179        4      15000 mmcblk0p4
 179        5        780 mmcblk0p5
 179        6        780 mmcblk0p6
 179        7        780 mmcblk0p7
 179        8       3072 mmcblk0p8
 179        9       1500 mmcblk0p9
 179       10       1500 mmcblk0p10
 179       11       2048 mmcblk0p11
 179       12       5120 mmcblk0p12
 179       13        512 mmcblk0p13
 179       14      16384 mmcblk0p14
 179       15        512 mmcblk0p15
 179       16          1 mmcblk0p16
 179       17       1500 mmcblk0p17
 179       18       2048 mmcblk0p18
 179       19       5120 mmcblk0p19
 179       20        512 mmcblk0p20
 179       21        512 mmcblk0p21
 179       22    1536000 mmcblk0p22
 179       23      74787 mmcblk0p23
 179       24       1024 mmcblk0p24
 179       25      16384 mmcblk0p25
 179       26          8 mmcblk0p26
 179       27          8 mmcblk0p27
 179       28          1 mmcblk0p28
 179       29         32 mmcblk0p29
 179       30     256000 mmcblk0p30
 179       31   12838863 mmcblk0p31
 
 
**********************************************************************************************************************************************


root@flo:/ # ls -la /dev/block/platform/msm_sdcc.1/                            
drwxr-xr-x system   root              2020-03-24 16:30 by-name
drwxr-xr-x root     root              2020-03-24 16:30 by-num
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0 -> /dev/block/mmcblk0
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p1 -> /dev/block/mmcblk0p1
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p10 -> /dev/block/mmcblk0p10
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p11 -> /dev/block/mmcblk0p11
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p12 -> /dev/block/mmcblk0p12
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p13 -> /dev/block/mmcblk0p13
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p14 -> /dev/block/mmcblk0p14
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p15 -> /dev/block/mmcblk0p15
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p16 -> /dev/block/mmcblk0p16
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p17 -> /dev/block/mmcblk0p17
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p18 -> /dev/block/mmcblk0p18
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p19 -> /dev/block/mmcblk0p19
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p2 -> /dev/block/mmcblk0p2
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p20 -> /dev/block/mmcblk0p20
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p21 -> /dev/block/mmcblk0p21
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p22 -> /dev/block/mmcblk0p22
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p23 -> /dev/block/mmcblk0p23
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p24 -> /dev/block/mmcblk0p24
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p25 -> /dev/block/mmcblk0p25
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p26 -> /dev/block/mmcblk0p26
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p27 -> /dev/block/mmcblk0p27
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p28 -> /dev/block/mmcblk0p28
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p29 -> /dev/block/mmcblk0p29
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p3 -> /dev/block/mmcblk0p3
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p30 -> /dev/block/mmcblk0p30
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p31 -> /dev/block/mmcblk0p31
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p4 -> /dev/block/mmcblk0p4
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p5 -> /dev/block/mmcblk0p5
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p6 -> /dev/block/mmcblk0p6
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p7 -> /dev/block/mmcblk0p7
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p8 -> /dev/block/mmcblk0p8
lrwxrwxrwx root     root              2020-03-24 16:30 mmcblk0p9 -> /dev/block/mmcblk0p9

**************************************************************************************************************


root@flo:/ # ls -la /dev/block/platform/msm_sdcc.1/by-n                    
by-name/  by-num/   
s -la /dev/block/platform/msm_sdcc.1/by-name                                  <
lrwxrwxrwx root     root              2020-03-24 16:30 DDR -> /dev/block/mmcblk0p26
lrwxrwxrwx root     root              2020-03-24 16:30 aboot -> /dev/block/mmcblk0p12
lrwxrwxrwx root     root              2020-03-24 16:30 abootb -> /dev/block/mmcblk0p19
lrwxrwxrwx root     root              2020-03-24 16:30 boot -> /dev/block/mmcblk0p14
lrwxrwxrwx root     root              2020-03-24 16:30 cache -> /dev/block/mmcblk0p23
lrwxrwxrwx root     root              2020-03-24 16:30 fsg -> /dev/block/mmcblk0p8
lrwxrwxrwx root     root              2020-03-24 16:30 m9kefs1 -> /dev/block/mmcblk0p5
lrwxrwxrwx root     root              2020-03-24 16:30 m9kefs2 -> /dev/block/mmcblk0p6
lrwxrwxrwx root     root              2020-03-24 16:30 m9kefs3 -> /dev/block/mmcblk0p7
lrwxrwxrwx root     root              2020-03-24 16:30 m9kefsc -> /dev/block/mmcblk0p28
lrwxrwxrwx root     root              2020-03-24 16:30 metadata -> /dev/block/mmcblk0p29
lrwxrwxrwx root     root              2020-03-24 16:30 misc -> /dev/block/mmcblk0p24
lrwxrwxrwx root     root              2020-03-24 16:30 modemst1 -> /dev/block/mmcblk0p2
lrwxrwxrwx root     root              2020-03-24 16:30 modemst2 -> /dev/block/mmcblk0p3
lrwxrwxrwx root     root              2020-03-24 16:30 pad -> /dev/block/mmcblk0p16
lrwxrwxrwx root     root              2020-03-24 16:30 persist -> /dev/block/mmcblk0p4
lrwxrwxrwx root     root              2020-03-24 16:30 radio -> /dev/block/mmcblk0p1
lrwxrwxrwx root     root              2020-03-24 16:30 recovery -> /dev/block/mmcblk0p25
lrwxrwxrwx root     root              2020-03-24 16:30 rpm -> /dev/block/mmcblk0p13
lrwxrwxrwx root     root              2020-03-24 16:30 rpmb -> /dev/block/mmcblk0p20
lrwxrwxrwx root     root              2020-03-24 16:30 sbl1 -> /dev/block/mmcblk0p9
lrwxrwxrwx root     root              2020-03-24 16:30 sbl2 -> /dev/block/mmcblk0p10
lrwxrwxrwx root     root              2020-03-24 16:30 sbl2b -> /dev/block/mmcblk0p17
lrwxrwxrwx root     root              2020-03-24 16:30 sbl3 -> /dev/block/mmcblk0p11
lrwxrwxrwx root     root              2020-03-24 16:30 sbl3b -> /dev/block/mmcblk0p18
lrwxrwxrwx root     root              2020-03-24 16:30 ssd -> /dev/block/mmcblk0p27
lrwxrwxrwx root     root              2020-03-24 16:30 system -> /dev/block/mmcblk0p22
lrwxrwxrwx root     root              2020-03-24 16:30 tz -> /dev/block/mmcblk0p15
lrwxrwxrwx root     root              2020-03-24 16:30 tzb -> /dev/block/mmcblk0p21
lrwxrwxrwx root     root              2020-03-24 16:30 userdata -> /dev/block/mmcblk0p31
lrwxrwxrwx root     root              2020-03-24 16:30 vendor -> /dev/block/mmcblk0p30

**Comment by ChiefPete** [comment]
After thinking about the size of the /system partition that I posted above, I realized that when you install a rom the partitions are not modified. - I had used Clamor's re-partition script https://forum.xda-developers.com/nexus-7-2013/orig-development/repartition-nexus-7-2013-repartition-t3844386 to increase the /system partition.

Thankfully he has a function built into the script to reset the partition sizes back to stock. After I reset the partitions back to stock, I was able to install Ubports.

Mark
